Add example of spectrum shifting to docs (#600)
astropy/specutils: docs/manipulation.rst

.. note:: The mask attribute is the only attribute modified by ``snr_threshold()``. To
             retrieve the masked flux data use ``spectrum.masked.flux_masked``.

Shifting
--------

> To shift a spectrum in wavelength, one can use a construct like this:

Reviewer: ```suggestion

In addition to resampling, you may sometimes wish to simply shift
the ``spectral_axis`` of a spectrum (a la the ``specshift`` iraf task).
There is no explicit function for this because it is a basic transform of
just the ``spectral_axis``.  Therefore one can use a construct like this:
```

Author: Done

.. code-block:: python

    >>> from specutils import Spectrum1D
    >>> wavelengths = np.arange(0, 10) * u.um
    >>> flux = 100 * np.abs(np.random.randn(10)) * u.Jy
    >>> spectrum = Spectrum1D(spectral_axis=wavelengths, flux=flux)

>     >>> spectrum  #doctest:+SKIP

Reviewer: why is this ``#doctest:+SKIP``? (same question for the ``new_spec`` line below)  I'd suggest instead maybe instead using ``FLOAT_CMP`` or ``ELLIPSES`` if the problem is floating-point precision.

Author: That's right. ELLIPSIS works better there.

    <Spectrum1D(flux=<Quantity [ 31.65751688, 115.6368968 , 118.58240303, 186.46500763,
                163.31955247,  90.92780989,  22.69088175,  51.81322025,
                 11.70522032,  63.81487901] Jy>, spectral_axis=<Quantity [0., 1., 2., 3.,
                 4., 5., 6., 7., 8., 9.] um>)>
    >>> shift = 12300 * u.AA

>     >>> new_spec = Spectrum1D(spectral_axis=spectrum.spectral_axis + shift, flux=spectrum.flux, wcs=None)

Reviewer: ```suggestion
    >>> new_spec = Spectrum1D(spectral_axis=spectrum.spectral_axis + shift, flux=spectrum.flux)
```
I don't think the ``wcs=None`` is necessary?

Author: It's not. I removed it. The idea was to make more clear to the reader that (for now) the internal wcs of the resulting spectrum wouldn't reflect the change made to the spectral axis. Maybe this should be said in the text itself?

.. note:: The mask attribute is the only attribute modified by ``snr_threshold()``. To
             retrieve the masked flux data use ``spectrum.masked.flux_masked``.

Shifting
--------

To shift a spectrum in wavelength, one can use a construct like this:

.. code-block:: python

    >>> from specutils import Spectrum1D
    >>> wavelengths = np.arange(0, 10)*u.um
    >>> flux = 100*np.abs(np.random.randn(10))*u.Jy
    >>> spectrum = Spectrum1D(spectral_axis=wavelengths, flux=flux)
    >>> >>> spectrum  #doctest:+SKIP
    <Spectrum1D(flux=<Quantity [ 31.65751688, 115.6368968 , 118.58240303, 186.46500763,
                163.31955247,  90.92780989,  22.69088175,  51.81322025,
                 11.70522032,  63.81487901] Jy>, spectral_axis=<Quantity [0., 1., 2., 3.,
                 4., 5., 6., 7., 8., 9.] um>)>
    >>> wunit = spectrum.spectral_axis.unit
    >>> shift = 1.23

>     >>> new_spec = Spectrum1D(spectral_axis=(spectrum.spectral_axis.value + shift) * wunit, flux=spectrum.flux, wcs=None)

Reviewer 2: I think this example should demonstrate using a shift as a `Quantity`. There's a case to be made that a shift could be applied to a spectral axis that is not in the same unit, adding the quantity version of the spectral axis with a quantity version of a shift should implicitly transform them to get the sum

```suggestion
    >>> shift = 12300 * u.AA
    >>> new_spec = Spectrum1D(spectral_axis=spectrum.spectral_axis + shift, flux=spectrum.flux, wcs=None)
```

Author: Done
